### PR TITLE
[Core] Python registry additions

### PR DESCRIPTION
**Pull request by rubenzorrilla**
**📝 Description**
This PR starts filling the Python registry in #10420. Thanks to the addition of the auxiliary methods in `python_registry_utilities.py` we only require the users to fill the corresponding `python_registry_lists.py`. So far I only added the `python_registry_lists.py` for the `KratosCore` and the `FluidDynamicsApplication` with a few examples. These are to be OFC extended in subsequent PRs.

In the long term, I think we should add the information in the `python_registry_lists.py` files to the json files at each application (i.e. `FluidDynamicsApplication.json`, `StructuralMechanicsApplication.json`, etc.). However, this would require to standarize the naming among applications, something that needs to be discussed. Nice thing is that, from the registry point of view, this change will be easily done under the hood.

Finally, I also note that the application generator has been updated accordingly.

### CODE REVIEW

File: kratos/templates/template_application/TemplateApplication.py.in [hunk 1]
```diff
@@ -1,7 +1,13 @@
...
+
 application = Kratos@{APP_NAME_CAMEL}Application()
 application_name = "Kratos@{APP_NAME_CAMEL}Application"
 
 _ImportApplication(application, application_name)
+
+from . import python_registry_lists
+python_registry_utilities.RegisterModelersList("KratosMultiphysics.@{APP_NAME_CAMEL}Application", python_registry_lists)
+python_registry_utilities.RegisterOperationsList("KratosMultiphysics.@{APP_NAME_CAMEL}Application", python_registry_lists)
+python_registry_utilities.RegisterProcessesList("KratosMultiphysics.@{APP_NAME_CAMEL}Application", python_registry_lists)
```
Comment (philbucher) [inline kratos/templates/template_application/TemplateApplication.py.in:13]:
```suggestion
python_registry_utilities.RegisterProcesses("KratosMultiphysics.@{APP_NAME_CAMEL}Application", python_registry)
```

more pythonic

Comment (rubenzorrilla) [reply kratos/templates/template_application/TemplateApplication.py.in:13]: You mean passing the module instead? If so, I've two concerns:
1. We already have a `python_registry` module
2. In the long run, the idea would be to substitute the `python_registry_lists` by json ones stored in the application.json files. If we do it in this way we won't require to change the implementations in the `python_registry_utilities.py`.

**Comment by pooyan-dadvand** [comment]
Online with our discussion! Thanks for the implementation.

Seeing the code and as you are not asking the list name for each item, I think we can substitute these three calls:

```Python
python_registry_utilities.RegisterModelersList("KratosMultiphysics.FluidDynamicsApplication", python_registry_lists)
python_registry_utilities.RegisterOperationsList("KratosMultiphysics.FluidDynamicsApplication", python_registry_lists)
python_registry_utilities.RegisterProcessesList("KratosMultiphysics.FluidDynamicsApplication", python_registry_lists)
```
with one call:
```Python
python_registry_utilities.ReadRegistryLists("KratosMultiphysics.FluidDynamicsApplication", python_registry_lists)
```

**Comment by rubenzorrilla** [comment]
> Online with our discussion! Thanks for the implementation.
> 
> Seeing the code and as you are not asking the list name for each item, I think we can substitute these three calls:
> 
> ```python
> python_registry_utilities.RegisterModelersList("KratosMultiphysics.FluidDynamicsApplication", python_registry_lists)
> python_registry_utilities.RegisterOperationsList("KratosMultiphysics.FluidDynamicsApplication", python_registry_lists)
> python_registry_utilities.RegisterProcessesList("KratosMultiphysics.FluidDynamicsApplication", python_registry_lists)
> ```
> 
> with one call:
> 
> ```python
> python_registry_utilities.ReadRegistryLists("KratosMultiphysics.FluidDynamicsApplication", python_registry_lists)
> ```


> Online with our discussion! Thanks for the implementation.
> 
> Seeing the code and as you are not asking the list name for each item, I think we can substitute these three calls:
> 
> ```python
> python_registry_utilities.RegisterModelersList("KratosMultiphysics.FluidDynamicsApplication", python_registry_lists)
> python_registry_utilities.RegisterOperationsList("KratosMultiphysics.FluidDynamicsApplication", python_registry_lists)
> python_registry_utilities.RegisterProcessesList("KratosMultiphysics.FluidDynamicsApplication", python_registry_lists)
> ```
> 
> with one call:
> 
> ```python
> python_registry_utilities.ReadRegistryLists("KratosMultiphysics.FluidDynamicsApplication", python_registry_lists)
> ```

Makes sense. Then we internally call the corresponding `RegisterModelersList`, `RegisterProcessesList` and so on. :+1:

**Review by pooyan-dadvand — APPROVED** [review]
Thanks @rubenzorrilla for the effort!

**Pull request merged**